cypress/integration/toolbar.spec.js

```diff
@@ -408,7 +408,28 @@ describe('Testing the Toolbar', () => {
     cy.toolbarButton('polygon').click();
 
     cy.window().then(() => {
-      expect(eventFired).to.not.equal('drawPolygon');
+      expect(eventFired).to.equal('drawPolygon');
+    });
+  });
+  it('Disable active button', () => {
+    let eventFired = '';
+    cy.window().then(({ map }) => {
+      map.on('pm:buttonclick', ({ btnName }) => {
+        eventFired = btnName;
+      });
+    });
+
+    cy.toolbarButton('polygon').click();
+
+    cy.window().then(({ map }) => {
+      map.pm.Toolbar.setButtonDisabled('drawPolygon', true);
+    });
+
+    cy.window().then(() => {
+      expect(eventFired).to.equal('drawPolygon');
+      cy.toolbarButton('polygon')
+        .closest('.button-container')
+        .should('have.not.class', 'active');
```

Comment:
Move this block above
```
 cy.window().then(({ map }) => {
      map.pm.Toolbar.setButtonDisabled('drawPolygon', true);
    });
```
And then add a check if the event / click happens after disableing the button

Reply:
Thank you, fixed.